packages/angular-cli/utilities/check-package-manager.ts

```diff
@@ -0,0 +1,60 @@
+import * as chalk from 'chalk';
+import {exec} from 'child_process';
+import {CliConfig} from '../models/config';
+
+const Promise = require('../ember-cli/lib/ext/promise');
+const config = CliConfig.fromProject();
+
+let packageManager = 'npm';
+if (config && config.get('packageManager')) {
+  packageManager = config.get('packageManager');
+}
+
+export function checkYarnOrCNPM() {
+  return new Promise((resolve, reject) => {
```

Comment:
No need to create a new promise (and it's kinda awkward). Just return the `Promise.all().then().catch()` directly.